systemd/systemd.go

```diff
@@ -767,25 +746,55 @@ LazyUnmount=yes
 
 [Install]
 WantedBy=multi-user.target
-`, snapName, revision, what, where, actualFsType, strings.Join(options, ","))
+`
 
+func writeMountUnitFile(snapName, revision, what, where, fstype string, options []string) (mountUnitName string, err error) {
+	content := fmt.Sprintf(mountUnitTemplate, snapName, revision, what, where, fstype, strings.Join(options, ","))
 	mu := MountUnitPath(where)
-	mountUnitName, err = filepath.Base(mu), osutil.AtomicWriteFile(mu, []byte(c), 0644, 0)
+	mountUnitName, err = filepath.Base(mu), osutil.AtomicWriteFile(mu, []byte(content), 0644, 0)
 	if err != nil {
-		return "", "", nil, err
+		return "", err
 	}
+	return mountUnitName, nil
+}
 
-	return mountUnitName, actualFsType, options, err
+func fsMountOptions(fstype string) []string {
+	options := []string{"nodev"}
+	if fstype == "squashfs" {
+		if selinux.ProbedLevel() != selinux.Unsupported {
+			if mountCtx := selinux.SnapMountContext(); mountCtx != "" {
+				options = append(options, "context="+mountCtx)
+			}
+		}
+	}
+	return options
+}
+
+func actualFsTypeAndMountOptions(fstype string) (actualFsType string, options []string) {
```

Comment:
I think this warrants a docstring that explains the difference.

Reading the entire patch I think the word _actual_ doesn't make sense here. Actual squash is a squash, we just choose to mount it with one or another implementation based on some conditions. Perhaps it would be better to have an explicit flag `UseFUSE` or something like that.

Reply:
Good idea, added a comment, thanks!

As for the second part, I'm not sure, we did use actual* before for variables that correspond to fstype in this context. I'm happy to reconsider naming in a followup if we have an agreement.

Reply:
Maybe something like `hostFsTypeAndMountOptions()` ?

Reply:
I like that, I'll do a followup (don't want to run spread again on this PR..)